src/Stratis.Features.FederatedPeg/Collateral/CheckCollateralFullValidationRule.cs

```diff
@@ -31,16 +32,16 @@ public class CheckCollateralFullValidationRule : FullValidationConsensusRule
         private readonly int collateralCheckBanDurationSeconds;
 
         public CheckCollateralFullValidationRule(IInitialBlockDownloadState ibdState, ICollateralChecker collateralChecker,
-            ISlotsManager slotsManager, IDateTimeProvider dateTime, Network network)
+            ISlotsManager slotsManager, IDateTimeProvider dateTime, Network network, CounterChainNetworkWrapper counterChainNetwork)
         {
-            this.network = network;
+            this.network = counterChainNetwork.CounterChainNetwork;
```

Comment:
`this.network` will need to be renamed as its confusing which one to use down below when calculating `collateralCheckBanDurationSeconds `

Reply:
Also on ln71 add a comment that `maxReorgLength` is a counter chain network length.

Reply:
Done